.github/workflows/rust.yml

```
@@ -310,15 +313,29 @@ jobs:
       - name: Run tests
         run: cargo test --test ${{ matrix.tests }} --manifest-path ${{ matrix.directory }}/Cargo.toml --no-fail-fast ${{ matrix.flags }}
 
-      - name: Read results
+      - name: Upload to DataDog
         if: always()
         run: |
-          find ${{ matrix.directory }}/${{ env.OUTPUT_DIRECTORY }} -name timing.json -exec cat {} + | jq '{(.test_path): . | del(.test_path)}' | jq -s add
+          message=$(find ${{ matrix.directory }}/${{ env.OUTPUT_DIRECTORY }} -name timing.json -exec cat {} + | jq '{(.test_path): . | del(.test_path)}' | jq -s add)
+          directory="${{ matrix.directory }}"
+          tags="crate:${directory##*/},event:${{ github.event_name }},test:${{ matrix.tests }},stage:dev,toolchain:${{ matrix.toolchain }},user:${{ github.actor }}"
+          if [[ ${{ github.ref_type }} = branch ]]; then
+            tags="$tags,branch:${{ github.ref_name }}"
+          else
+            tags="$tags,tag:${{ github.ref_name }}"
```

I would perhaps change this just to be a little clearer
```suggestion
            tags="$tags,git_tag:${{ github.ref_name }}"
```

---

I'm rethinking this part. Currently, we don't have any releases and we don't plan to add a tag soon-ish. Let's just remove it?
When we do releases, I think, we want a persistent tag `version:*.*.*`?

---

sounds good